.github/workflows/pr-labeler.yml

```
@@ -1,6 +1,7 @@
 name: "Pull Request Labeler"
 on:
-- pull_request_target
+  pull_request:
+    types: [opened]

```

```suggestion
    types: [opened, edited, reopened]
```

I imagine as PRs change new labels might become relevant. What do you think about using these 3 types?

Linking [the docs here](https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#pull_request_target) since it was hard for me to find them.

---

It's unclear from that list which type is "force pushed to branch" or "pushed to branch" both of which would be great to _not_ run the action again.

---

So, what I would like to do is find a way to strike a balance between how the actions run. @rboyer ran into a situation where it was reapplying on every push, and in some cases, inaccurately.

I think moving away from `pull_request_target` was the first step. Here's the description:
>By default, a workflow only runs when a pull_request_target's activity type is opened, synchronize, or reopened. To trigger workflows for more activity types, use the types keyword.

I think moving it to opened, reopened is a good idea, but the "Edited" may put us back in this scenario. What about **unlabeled** so that if there is no label when the PR is opened, we should be safe to add. And then once it's labeled, it shouldn't trigger.

We could look at making some conditional [expressions](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions) within the syntax, but I think that would gate the whole workflow.

---

Where the incorrect labels added because it was an edit, or because the rules were wrong? I do not understand why running on edit would be a problem, assuming it always uses the same rules.

---

If we can make it conditional on a specific label we could use a `disable-labeler` label. But maybe we wait until we've sorted out the rules some more.